haystack/nodes/preprocessor/preprocessor.py

```diff
@@ -169,8 +172,26 @@ def process(
         else:
             raise Exception("documents provided to PreProcessor.prepreprocess() is not of type list nor Document")
 
+        self._long_documents(ret, max_chars=self.max_chars)
+
         return ret
 
+    def _long_documents(self, documents: List[Document], max_chars=10_000):
+        """
+        Function that tries to detect unusually long documents.
+
+        NOTE: this function is a heuristic that is in place only because a proper fix that prevents such documents from forming
+        would imply a complete revamp of this class, including better definitions of what the various units (word, sentence, passage) mean exactly.
+        """
+        for document in documents:
+            if len(document.content) > max_chars:
+                logger.warning(
+                    "Document %s is %s characters long after preprocessing, where the maximum lenght should be %s.",
```

Comment:
1. "lenght" -> "length"
2. Do you think it's clear enough that nothing's actually been _done_ about the max length being exceeded? Or do you think the fact that it's just a warning conveys this adequately?

Reply:
Maybe we should consider renaming the variable to `max_chars_check`, so it's more clear that this variable is only used for checking the max number of chars?

Reply:
I've improved a bit the error message as well